core/src/env_platform/env_service.bal

```diff
@@ -61,6 +61,23 @@ service envservice on ep0 {
     }
     resource function postApplication(http:Caller caller, http:Request req, TreeRemovalForm body) returns error? {
 
+        http:Response response = new;
+        string authHeader = req.getHeader("Authorization");
+        [string, string]|error userInfoFromJWT = getUserInfoFromJWT(authHeader);
+        if (userInfoFromJWT is [string, string]) {
+            [string, string] [userId, userType] = userInfoFromJWT;
+            [boolean, string]|error saveApplicationResult = saveApplication(body, userId);
+            if (saveApplicationResult is error) {
+                response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+            } else {
+                [boolean, string] [isSaved, applicationId] = saveApplicationResult;
+                response.statusCode = http:STATUS_CREATED;
+                response.setPayload(<@untainted>{"applicationId": applicationId});
+            }
+        } else {
+            response.statusCode = http:STATUS_UNAUTHORIZED;
```

Comment:
Same as the above comment, let's have a payload containing the error message **"Unauthorized operation. Try again with valid credentials."** for this.